addon/-private/system/store.js

```diff
@@ -2107,7 +2114,11 @@ Store = Service.extend({
     let trueModelName = this._classKeyFor(modelName);
     let owner = getOwner(this);
 
-    return owner._lookupFactory(`model:${trueModelName}`);
+    if (owner.factoryFor) {
+      return owner.factoryFor(`model:${trueModelName}`) && owner.factoryFor(`model:${trueModelName}`).class;
```

Comment:
Same comment as above, we shouldn't call `factorFor` twice in a row like this...